VSRAD.Package/ProjectSystem/SolutionProperties.cs

```diff
@@ -0,0 +1,61 @@
+﻿using EnvDTE;
+using Microsoft.VisualStudio.ProjectSystem;
+using Microsoft.VisualStudio.Shell;
+using System.ComponentModel;
+using System.ComponentModel.Composition;
+using System.IO;
+using VSRAD.Package.Options;
+
+namespace VSRAD.Package.ProjectSystem
+{
+    [Export]
+    public sealed class SolutionProperties
+    {
+        public ProjectOptions Options { get; private set; }
+        private string _optionsFilePath;
+
+        public void SetOptions(Solution solution, UnconfiguredProject project)
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
+
+            var solutionPath = solution.FullName;
+            var legacyOptionsPath = project.FullPath + ".conf.json";
+            var legacyOptionsPathAternative = project.FullPath + ".user.json";
+            var solutionConfigPath = !string.IsNullOrWhiteSpace(solutionPath) ? GetConfigPath(solutionPath) : "";
+
+            Options = !string.IsNullOrWhiteSpace(solutionPath) && File.Exists(solutionConfigPath)
+                ? ProjectOptions.Read(solutionConfigPath)
+                : File.Exists(legacyOptionsPath)
+                    ? ProjectOptions.Read(legacyOptionsPath)
+                    : File.Exists(legacyOptionsPathAternative)
+                        ? ProjectOptions.ReadLegacy(legacyOptionsPathAternative)
+                        : new ProjectOptions();
+
+            _optionsFilePath = string.IsNullOrWhiteSpace(solutionPath)
```

Comment:
Here is a repetition of `string.IsNullOrWhiteSpace(solutionPath)`. It would be better to add local variable `var isSolutionExists = string.IsNullOrWhiteSpace(solutionPath);` which also improves understanding of the code.

Reply:
Yep, resolved in eff6af0